PORTAL/jobs/queue.py

```diff
@@ -345,3 +384,21 @@ def read_log(self) -> Iterator[_utils.LogSection]:
             return
         with logfile:
             yield from _utils.LogSection.read_logfile(logfile)
+
+
+class JobQueues:
+    @classmethod
+    def from_config(cls, cfg: "JobsConfig", fs: JobQueuesFS):
+        return cls(cfg, fs)
+
+    def __init__(self, cfg: "JobsConfig", fs: JobQueuesFS):
+        self._cfg = cfg
+        self._fs = fs
+
+    def get_queue(self, queueid: str) -> JobQueue:
+        assert queueid in self._cfg.workers, queueid
+        return JobQueue.from_fstree(self._fs.resolve_queue(queueid))
+
+    def iter_queues(self) -> Iterator[JobQueue]:
+        for queueid in self._cfg.workers.keys():
+            yield JobQueue.from_fstree(self._fs.resolve_queue(queueid))
```

Comment:
FWIW, having "queue" in the method names is a bit redundant.

```suggestion
    def __iter__(self) -> Iterator[JobQueue]:
        for queueid in self._cfg.workers.keys():
            yield JobQueue.from_fstree(self._fs.resolve_queue(queueid))

    def get(self, queueid: str) -> JobQueue:
        assert queueid in self._cfg.workers, queueid
        return JobQueue.from_fstree(self._fs.resolve_queue(queueid))
```

or even

```suggestion
    def __len__(self):
        return len(self._cfg.workers)

    def __iter__(self) -> Iterator[JobQueue]:
        for queueid in self._cfg.workers.keys():
            yield JobQueue.from_fstree(self._fs.resolve_queue(queueid))

    def __getitem__(self, queueid: str) -> JobQueue:
        assert queueid in self._cfg.workers, queueid
        return JobQueue.from_fstree(self._fs.resolve_queue(queueid))
```